src/framework/theme/components/autocomplete/autocomplete.component.spec.ts

```diff
@@ -0,0 +1,273 @@
+import { Component, EventEmitter, Output, QueryList, ViewChild, ViewChildren } from '@angular/core';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { RouterTestingModule } from '@angular/router/testing';
+import { FormsModule, ReactiveFormsModule } from '@angular/forms';
+import { NB_DOCUMENT } from '@nebular/theme/theme.options';
+import { By } from '@angular/platform-browser';
+import {
+  NbOptionComponent,
+  NbAutocompleteModule,
+  NbOptionListModule,
+  NbThemeModule,
+  NbLayoutModule,
+  NbOverlayContainerAdapter,
+} from '@nebular/theme';
+import { NbAutocompleteComponent } from './autocomplete.component';
+import { NbAutocompleteDirective } from './autocomplete.directive';
+
+const TEST_GROUPS = [
+  {
+    title: 'Group 1',
+    options: [
+      { title: 'Option 1', value: 'Option 1' },
+      { title: 'Option 2', value: 'Option 2' },
+      { title: 'Option 3', value: 'Option 3' },
+    ],
+  },
+  {
+    title: 'Group 2',
+    options: [
+      { title: 'Option 21', value: 'Option 21' },
+      { title: 'Option 22', value: 'Option 22' },
+      { title: 'Option 23', value: 'Option 23' },
+    ],
+  },
+  {
+    title: 'Group 3',
+    options: [
+      { title: 'Option 31', value: 'Option 31' },
+      { title: 'Option 32', value: 'Option 32' },
+      { title: 'Option 33', value: 'Option 33' },
+    ],
+  },
+];
+
+@Component({
+  selector: 'nb-autocomplete-test',
+  template: `
+    <nb-layout>
+      <nb-layout-column>
+
+        <input #autoInput
+               nbInput
+               type="text"
+               (input)="onChange($event)"
+               placeholder="This is test autocomplete component"
+               [nbAutocomplete]="auto" />
+
+        <nb-autocomplete #auto (selectedChange)="selectedChange.emit($event)" [handleDisplayFn]="handleFunction">
+
+          <nb-option-group *ngFor="let group of filteredGroups" [title]="group.title">
+            <nb-option *ngFor="let option of group.options" [value]="option.value">
+              {{ option.title }}
+            </nb-option>
+          </nb-option-group>
+
+        </nb-autocomplete>
+
+      </nb-layout-column>
+    </nb-layout>
+  `,
+})
+export class NbAutocompleteTestComponent {
+
+  @Output() selectedChange: EventEmitter<any> = new EventEmitter();
+  @ViewChild(NbAutocompleteDirective, { static: true }) autocompleteDirective: NbAutocompleteDirective<string>;
```

Comment:
Is `{ static: true }` required here? Afaik, static resolution of children won't be available in Angular 9, so we would have to remove this soon anyway

Reply:
Changed to `{ static: false }`.